minecraft/common/src/main/java/dev/kosmx/playerAnim/mixin/CapeLayerMixin.java

```diff
@@ -1,29 +1,71 @@
     private void render(PoseStack poseStack, MultiBufferSource multiBufferSource, int i, AbstractClientPlayer abstractClientPlayer, float f, float g, float h, float j, float k, float l, CallbackInfo ci) {
         AnimationApplier emote = ((IAnimatedPlayer) abstractClientPlayer).playerAnimator_getAnimation();
-        Vec3f pos = emote.get3DTransform("torso", TransformType.POSITION, Vec3f.ZERO);
-        poseStack.translate(pos.getX()/16, pos.getY()/16, pos.getZ()/16);
-        Vec3f rot = emote.get3DTransform("torso", TransformType.ROTATION, Vec3f.ZERO);
-        poseStack.mulPose(Axis.ZP.rotation(rot.getZ()));    //roll
-        poseStack.mulPose(Axis.YP.rotation(rot.getY()));    //pitch
-        poseStack.mulPose(Axis.XP.rotation(rot.getX()));    //yaw
+        if (emote.isActive()) {
+            ModelPart torso = this.getParentModel().body;
+
+            poseStack.rotateAround((new Quaternionf()).rotateXYZ(torso.xRot, torso.yRot, torso.zRot), torso.x/16, torso.y/16, torso.z/16);
+
+            poseStack.translate(0.0F, 0.0F, 0.125F);
+            poseStack.translate(torso.x / 16, torso.y / 16, torso.z / 16);
+
+            double d = Mth.lerp(h, abstractClientPlayer.xCloakO, abstractClientPlayer.xCloak)
+                    - Mth.lerp(h, abstractClientPlayer.xo, abstractClientPlayer.getX());
+            double m = Mth.lerp(h, abstractClientPlayer.zCloakO, abstractClientPlayer.zCloak)
+                    - Mth.lerp(h, abstractClientPlayer.zo, abstractClientPlayer.getZ());
+            float n = Mth.rotLerp(h, abstractClientPlayer.yBodyRotO, abstractClientPlayer.yBodyRot);
+            double o = Mth.sin(n * (float) (Math.PI / 180.0));
+            double p = -Mth.cos(n * (float) (Math.PI / 180.0));
+            float s = (float) (d * p - m * o) * 100.0F;
+            s = Mth.clamp(s, -20.0F, 20.0F);
+
+            poseStack.mulPose(Axis.XP.rotationDegrees(6.0F / 2.0F));
+            poseStack.mulPose(Axis.ZP.rotationDegrees(s / 2.0F));
+            poseStack.mulPose(Axis.YP.rotationDegrees(180.0F - s / 2.0F));
+        }
+    }
+
+    @Redirect(method = "render(Lcom/mojang/blaze3d/vertex/PoseStack;Lnet/minecraft/client/renderer/MultiBufferSource;ILnet/minecraft/client/player/AbstractClientPlayer;FFFFFF)V", at = @At(value = "INVOKE", target = "Lcom/mojang/blaze3d/vertex/PoseStack;mulPose(Lorg/joml/Quaternionf;)V"))
+    private void mulPose(PoseStack instance, Quaternionf quaternionf, @Local(argsOnly = true) AbstractClientPlayer abstractClientPlayer) {
+        AnimationApplier emote = ((IAnimatedPlayer) abstractClientPlayer).playerAnimator_getAnimation();
+        if (!emote.isActive()) {
+            instance.mulPose(quaternionf);
```

Comment:
I think it's better something like:

https://github.com/LlamaLad7/MixinExtras/wiki/WrapWithCondition
https://github.com/LlamaLad7/MixinExtras/wiki/WrapOperation

Reply:
indeed!